File: src/main/java/demo/helloworld/HelloWorldApp.java
package demo.helloworld;

import act.Act;
import act.inject.DefaultValue;
import act.util.Output;
import org.osgl.mvc.annotation.GetAction;

import static act.controller.Controller.Util.render;

public class HelloWorldApp {

    @GetAction
    public void home(@DefaultValue("World") @Output String who) {
        render(who);
    }

    public static void main(String[] args) throws Exception {
        Act.start();
    }

}

File: src/main/kotlin/demo/helloworld/hello-world-app.kt
package demo.helloworld

import act.Act
import act.inject.DefaultValue
import act.util.Output
import org.osgl.mvc.annotation.GetAction

import act.controller.Controller.Util.render

class HelloWorldApp {

    @GetAction
    fun home(@DefaultValue("World") @Output who: String) {
        render(who)
    }

    companion object {

        @Throws(Exception::class)
        @JvmStatic
        fun main(args: Array<String>) {
            Act.start()
        }
    }

}
